core/consensus/babe/impl/babe_config_repository_impl.cpp

```diff
@@ -86,49 +91,62 @@ namespace kagome::consensus::babe {
   }
 
   bool BabeConfigRepositoryImpl::prepare() {
-    if (auto res = persistent_storage_->tryGet(
-            storage::kBabeConfigRepositoryImplGenesisSlot)) {
+    if (auto res = persistent_storage_->tryGet(storage::kFirstBlockSlot)) {
       if (auto &genesis_slot_raw = res.value()) {
         if (auto res = scale::decode<SlotNumber>(*genesis_slot_raw)) {
           first_block_slot_number_ = res.value();
         } else {
           SL_ERROR(logger_, "genesis slot decode error: {}", res.error());
-          std::ignore = persistent_storage_->remove(
-              storage::kBabeConfigRepositoryImplGenesisSlot);
+          std::ignore = persistent_storage_->remove(storage::kFirstBlockSlot);
         }
       }
     } else {
       SL_ERROR(logger_, "genesis slot db read error: {}", res.error());
       return false;
     }
-    std::unique_lock lock{indexer_mutex_};
-    auto finalized = block_tree_->getLastFinalized();
-    auto finalized_header = block_tree_->getBlockHeader(finalized.hash).value();
 
-    if (finalized.number - indexer_.last_finalized_indexed_.number
-            > kMaxUnindexedBlocksNum
-        and trie_storage_->getEphemeralBatchAt(finalized_header.state_root)) {
-      warp(lock, finalized);
-    }
+    {
```

Comment:
Remove unnecessary scope.
Also `mutex` required for `if (dynamic_cast<Babe>) config()` below.
```suggestion
```